src/main/java/org/example/services/UserManager.java

```diff
@@ -0,0 +1,51 @@
+package org.example.services;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import org.example.commands.AddUserCommand;
+import org.example.commands.DeleteUserCommand;
+import org.example.commands.ProgramTerminateCommand;
+import org.example.commands.SaveUserCommand;
+import org.example.commands.SortUserCommand;
+import org.example.commands.UserCommandInterface;
+import org.example.discstorage.UserInfoDiscStorageHandler;
+import org.example.model.User;
+
+public class UserManager implements UserManagerInterface {
+  public Set<User> users;
+  private final UserInteractionServiceInterface userInteractionService;
+  private final Map<Integer, UserCommandInterface> commandMap;
+  private boolean terminateProgram = false;
+
+  public UserManager(UserInteractionServiceInterface userInteractionService,
+                     AddUserDetailsServiceInterface addUserDetails,
+                     SortUserDetailsServiceInterface sortUserDetailsService,
+                     DeleteUserDetailsServiceInterface deleteUserDetails,
+                     UserInfoDiscStorageHandler userInfoDiscStorageHandler) {
+    this.users = new HashSet<>();
+    this.users = userInfoDiscStorageHandler.getUsersFromDisk();
+    this.userInteractionService = userInteractionService;
+    this.commandMap = new HashMap<>();
+    commandMap.put(1, new AddUserCommand(addUserDetails));
```

Comment:
very good use of command design pattern 🙌

Reply:
thanks 😊